src/com/github/johnthagen/cppcheck/CppcheckInspection.java

```diff
@@ -37,122 +42,135 @@
-        // Example:
-        //   Checking Test.cpp ...
-        //   [Test.h:2]: (style) Unused variable: x
-        //   [Test.cpp:3]: (style) Unused variable: y
-        if (!fileName.equals(file.getName())) {
-          continue;
-        }
-
-        // Because cppcheck runs on physical files, it's possible for the editor lines
-        // (lines in the IDE memory) to get out of sync from the lines on disk.
-        if (lineNumber > document.getLineCount()) {
-          lineNumber = document.getLineCount();
-        }
-
-        // Cppcheck error or parsing error.
-        if (lineNumber <= 0) {
-          continue;
-        }
-
-        // Document counts lines starting at 0, rather than 1 like in cppcheck.
-        lineNumber -= 1;
-
-        final int lineStartOffset = document.getLineStartOffset(lineNumber);
-        final int lintEndOffset = document.getLineEndOffset(lineNumber);
-
-        // Do not highlight empty whitespace prepended to lines.
-        String line_text = document.getImmutableCharSequence().subSequence(
-          lineStartOffset, lintEndOffset).toString();
-
-        final int numberOfPrependedSpaces = line_text.length() -
-                                            line_text.replaceAll("^\\s+","").length();
-
-        ProblemDescriptor problemDescriptor = manager.createProblemDescriptor(
-          file,
-          TextRange.create(lineStartOffset + numberOfPrependedSpaces, lintEndOffset),
-          "cppcheck: (" + severity + ") " + errorMessage,
-          severityToHighlightType(severity),
-          true);
-        descriptors.add(problemDescriptor);
+      if (!cppcheckOutput.isEmpty()) {
+        List<ProblemDescriptor> descriptors = parseOutput(file, manager, document, cppcheckOutput, tempFile.getName());
+        return descriptors.toArray(new ProblemDescriptor[0]);
       }
-    } catch (ExecutionException ex) {
+    } catch (ExecutionException | IOException ex) {
       Notifications.Bus.notify(new Notification("cppcheck",
                                                 "Error",
-                                                "ExecutionException: " + ex.getMessage(),
+                                                ex.getClass().getSimpleName() + ": " + ex.getMessage(),
                                                 NotificationType.INFORMATION));
       ex.printStackTrace();
+    } finally {
+      if (tempFile != null) {
+        FileUtil.delete(tempFile);
+      }
     }
 
-    return descriptors.toArray(new ProblemDescriptor[descriptors.size()]);
+    return ProblemDescriptor.EMPTY_ARRAY;
+  }
+
+  @NotNull
+  private static String prependIncludeDir(@NotNull String cppcheckOptions, @NotNull VirtualFile vFile) {
+    VirtualFile dir = vFile.getParent();
+    if (dir == null) return cppcheckOptions;
+    String path = dir.getCanonicalPath();
+    if (path == null) return cppcheckOptions;
+    return String.format("-I\"%s\" %s", path, cppcheckOptions);
```

Comment:
What happens if the user provides their own `-I` options? Will this conflict? If so, do we need to at least add a mention about that in the configuration window to warn users?

Reply:
If I read cppcheck docs correctly, multiple `-I`s are allowed, and they are searched in the order they appear (so that's why I'm adding it in front: current dir always takes precedence in the header search path).

Btw, it lacks `""` vs` <>` differentiation, so now `<>`-path could be found relative to the original file's directory. I don't know if it's possible to fix without changing the cppcheck itself, and I think the benefits outweigh the possible problems.